sdks/go/pkg/beam/transforms/sql/sql_test.go

```diff
@@ -0,0 +1,105 @@
+package sql
+
+import (
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/transforms/sql/sqlx"
+	"reflect"
+	"runtime"
+	"testing"
+)
+
+func TestOptions_Add(t *testing.T) {
+	o := options{}
+	str := "this is a string"
+	bytes := []byte{1, 2, 3, 4}
+	testOpt := sqlx.Option{
+		Urn:     str,
+		Payload: bytes,
+	}
+	tests := []struct {
+		opt sqlx.Option
+	}{
+		{
+			opt: testOpt,
```

Comment:
Since there's only one case, this test becomes more readable if we avoid the table test loop. 

Alternatively, prefer inlining the definition of testOpt into the tests slice so case by case details don't leak.

Reply:
Thanks for reviewing my PR; I have changed my code.

The reason I originally used the tests slice is that it may support more extensibility if we later add more test cases. But since just one test case can successfully test the function, there is no need for more test cases, I think.